### PR TITLE
Proposal: Implement Ceph rbd storage driver

### PR DESCRIPTION
**Pull request by hustcat**
Long times ago, @dachary proposed implement [Ceph storage driver](https://github.com/docker/docker/issues/8854), but still not achieved now. I initially realized ceph rbd storage driver, and I hope this is the beginning of implement rbd storage driver .
@leseb @simon3z 
Signed-off-by: Ye Yin eyniy@qq.com

### CODE REVIEW

**Comment by cpuguy83** [comment]
This was originally proposed and closed here: https://github.com/docker/docker/pull/9146#issuecomment-67211228

Has anything changed?

File: daemon/graphdriver/rbd/README.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+---> Making bundle: dynbinary (in bundles/1.8.0-dev/dynbinary)
+Created binary: bundles/1.8.0-dev/dynbinary/dockerinit-1.8.0-dev
+Building: bundles/1.8.0-dev/dynbinary/docker-1.8.0-dev
+Created binary: bundles/1.8.0-dev/dynbinary/docker-1.8.0-dev
+```
+# How to use
+
+## install ceph cluster
+TODO:
```
Comment (leseb) [inline daemon/graphdriver/rbd/README.md]: Perhaps you get suggest to use the ceph/demo container for testing. https://github.com/ceph/ceph-docker/tree/master/demo

**Comment by hustcat** [comment]
@cpuguy83 Yea, the basic principle is almost same.However I use [go-ceph](https://github.com/noahdesu/go-ceph) as client library.I hope to promote docker and ceph community together to achieve rbd storage driver.This is  a great demand for many people.

File: daemon/graphdriver/rbd/README.md (hunk 1 above)
Comment (hustcat) [inline daemon/graphdriver/rbd/README.md]: Thanks. I will complete it later:)

**Comment by cpuguy83** [comment]
It looks like go-ceph is still just pulling in the C libs which seems like is the problem we had in the previous attempt.

**Comment by hustcat** [comment]
Yes, I wish ceph community can solve it as soon as possible.

**Comment by cpuguy83** [comment]
@hustcat Given that, I don't see a reason to keep this open, unfortunately... wdyt?
I would however like to point you to https://github.com/docker/docker/pull/13777 which would enable out of process graph drivers.

**Comment by hustcat** [comment]
@cpuguy83 Thanks, out of process graph drivers is a better choice so far:)

**Comment by hustcat** [comment]
Ceph rbd storage driver plugin refer to #13777, see [docker-rbd-driver](https://github.com/hustcat/docker-rbd-driver)

**Pull request closed without merging**